diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf12979..0cb14a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,9 +89,9 @@ else(DEFINED MUST_BUILD_CURL_CLIENT)
   set(tmp)
 endif(DEFINED MUST_BUILD_CURL_CLIENT)
 
-find_package(CURL ${tmp})
+#find_package(CURL ${tmp})
 
-ensc_set_bool(MUST_BUILD_CURL_CLIENT ${CURL_FOUND} "Set if Curl client transport shall be built")
+ensc_set_bool(MUST_BUILD_CURL_CLIENT 1 "Set if Curl client transport shall be built")
 set(curl_srcdir ${xmlrpc-c_SOURCE_DIR}/lib/curl_transport)
 
 if(MUST_BUILD_CURL_CLIENT)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index efb03e4..19226c8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -26,10 +26,10 @@ if(MUST_BUILD_LIBWWW_CLIENT)
 endif(MUST_BUILD_LIBWWW_CLIENT)
 
 if(MUST_BUILD_CURL_CLIENT)
-  FIND_PACKAGE(CURL REQUIRED)
-  include_directories(${CURL_INCLUDE_DIR})
-  set(client_curl_LIBS ${CURL_LIBRARY})
-  set(xmlrpc_client_pkgconfig_req libcurl)
+  #FIND_PACKAGE(CURL REQUIRED)
+  #include_directories(${CURL_INCLUDE_DIR})
+  #set(client_curl_LIBS ${CURL_LIBRARY})
+  #set(xmlrpc_client_pkgconfig_req libcurl)
   list(APPEND transport_SOURCES
     ${curl_srcdir}/xmlrpc_curl_transport.c
     ${curl_srcdir}/curltransaction.c
@@ -75,9 +75,15 @@ add_library(xmlrpc_client
   ${xmlrpc-c_SOURCE_DIR}/include/xmlrpc-c/client.h)
 
 if(WIN32)
-   target_link_libraries(xmlrpc_client xmlrpc ${client_LIBS} wldap32)
+   target_link_libraries(xmlrpc_client
+   PUBLIC xmlrpc ${client_LIBS} wldap32
+   PRIVATE  $<BUILD_INTERFACE:libcurl>
+   )
 else(WIN32)
-    target_link_libraries(xmlrpc_client xmlrpc ${client_LIBS})
+   target_link_libraries(xmlrpc_client
+   PUBLIC xmlrpc ${client_LIBS}
+   PRIVATE $<BUILD_INTERFACE:libcurl>
+   )
 endif(WIN32)
 set_target_properties(xmlrpc_client
   PROPERTIES
